examples/basecoin/app/init_baseapp.go

```diff
@@ -26,3 +33,68 @@ func (app *BasecoinApp) initBaseAppTxDecoder() {
 		return tx, nil
 	})
 }
+
+// We use GenesisAccount instead of types.AppAccount for cleaner json input of PubKey
+type GenesisAccount struct {
+	Name     string         `json:"name"`
+	Address  crypto.Address `json:"address"`
+	Coins    sdk.Coins      `json:"coins"`
+	PubKey   cmn.HexBytes   `json:"public_key"`
+	Sequence int64          `json:"sequence"`
+}
+
+func NewGenesisAccount(aa *types.AppAccount) *GenesisAccount {
+	return &GenesisAccount{
+		Name:     aa.Name,
+		Address:  aa.Address,
+		Coins:    aa.Coins,
+		PubKey:   aa.PubKey.Bytes(),
+		Sequence: aa.Sequence,
+	}
+}
+
+// convert GenesisAccount to AppAccount
+func (ga *GenesisAccount) toAppAccount() (acc *types.AppAccount, err error) {
+
+	pk, err := crypto.PubKeyFromBytes(ga.PubKey)
+	if err != nil {
+		return
+	}
+	baseAcc := auth.BaseAccount{
+		Address:  ga.Address,
+		Coins:    ga.Coins,
+		PubKey:   pk,
+		Sequence: ga.Sequence,
+	}
+	return &types.AppAccount{
+		BaseAccount: baseAcc,
+		Name:        ga.Name,
+	}, nil
+}
+
+// define the custom logic for basecoin initialization
+func (app *BasecoinApp) initBaseAppInitStater() {
+	accountMapper := app.accountMapper
+
+	app.BaseApp.SetInitStater(func(ctx sdk.Context, state json.RawMessage) sdk.Error {
+		if state == nil {
+			return nil
+		}
+
+		var gaccs []*GenesisAccount
```

Comment:
we should more clearly state this expectation somewhere - probably define an AppGenesis type that is a `[]*GenesisAccount`  ... or probably better to use a struct containing a []*GenesisAccount. 

ie. we want to make clear somewhere outside this function what the json in the genesis should look like

Reply:
good point, an alternative design should actually be used for custom genesis types which defines its own unmarshalling. aka AppAccount should actually exist as a custom module within basecoin which defines genesis unmarshalling... right now everything is just kind of unmarshalled here, I'm going to integrate this change in with the basecoin refactor